modules/@apostrophecms/image/ui/apos/components/AposMediaManager.vue

```diff
@@ -547,14 +551,17 @@ export default {
       this.skipLoadObserver = false;
     },
 
-    async onContentChanged({ action, doc }) {
-      if (doc.type !== '@apostrophecms/image' || ![ 'archive', 'update' ].includes(action)) {
+    async onContentChanged({
+      action, doc, moduleType
```

Comment:
Did you do this moduleType work to allow multiple subclasses of image pieces? I don't think that was on anyone's radar...

Reply:
Nope, I did it because we use the same event `content-changed` for every pieces, pages...
while when acting on one document is ok because we pass the document which contains `type` and can be checked to know if we should reload data in this specific modal.
We had no information about type for batch operations, just an array of IDS. Which means that the pages manager modal would reload data if I perform a batch operation on image or pieces, which makes no sense.

but the work isn't finished it, it should be used in page and piece managers too.